boot/core/src/boot/task/built_in.clj

```diff
@@ -479,6 +479,28 @@
           (core/with-post-wrap [_]
             (when (or client (not server)) @repl-cli)))))
 
+(core/deftask socket-server
+  "Start a socket server.
+
+  The default behavior is to serve a simple Clojure REPL. To serve a different
+  handler function, specify a symbol using `--accept'.
+
+  If no bind address is specified, the socket server will listen on 127.0.0.1.
+
+  If no port is specified, an open port will be chosen automatically. The port
+  number is written to .socket-port in the current directory.
+
+  The REPL can be accessed with the command
+
+     $ nc localhost $(cat .server-port)"
+
+  [b bind ADDR      str    "The address server listens on."
+   p port PORT      int    "The port to listen to."
+   a accept ACCEPT  sym    "Namespaced symbol of the accept function to invoke."]
+  (let [repl-soc (delay (core/launch-socket-server *opts*))]
+    (core/with-pass-thru [fs]
+      @repl-soc)))
```

Comment:
Cleanup task is currently problematic with REPL. If `boot` is run inside repl task, cleanup from that boot-in-boot would kill also host REPL. So, for now, this probably shouldn't use `cleanup`.